### PR TITLE
Twitterding websockets

### PR DESCRIPTION
**Pull request by feliciaan**

### CODE REVIEW

File: Twitter/boxxy.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+    def position(self, position):
+        pass
+
+    def dump(self):
+        with open(FILENAME, 'w') as f:
+            print(jp.encode(self), end='', file=f)
+
+auth = OAuth(ACCESS_KEY, ACCESS_SECRET, CONSUMER_KEY, CONSUMER_SECRET)
+twitter = Twitter(auth=auth)
```
Comment (FelixVanderJeugt) [inline Twitter/boxxy.py]: why is this not in main()?

File: Twitter/boxxy.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
+from socketIO_client import SocketIO, LoggingNamespace
+from functools import wraps
+from datetime import datetime, timedelta
+from collections import deque
+from random import choice
+from os.path import isfile
+from twitter import Twitter, OAuth
+import jsonpickle as jp
+
+from config import *
```
Comment (FelixVanderJeugt) [inline Twitter/boxxy.py]: Please fix.

File: Twitter/boxxy.py [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+        "/ping": boxxy.ping,
+        "/state": boxxy.state,
+        "/lap": boxxy.lap,
+        "/position": boxxy.position
+    }
+    for path, method in paths.items():
+        socketIO.on(path, method)
+
+    socketIO.wait(seconds=3600 * 12)
+
```
Comment (Silox) [inline Twitter/boxxy.py]: PEP8

File: Twitter/boxxy.py [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+        self.lastLapTimeStamp = datetime.utcnow() - timedelta(hours=1)  # be certain
+
+
+class Boxxy(object):
+    def __init__(self):
+        self.teams = {}
+        self.shortestLapGlobal = None
+        self.omtrek = float("inf")
+        self.distances = DISTANCES
+
```
Comment (Silox) [inline Twitter/boxxy.py]: PEP8

File: Twitter/boxxy.py [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+    def ping(self, json):
+        pass
+
```
Comment (Silox) [inline Twitter/boxxy.py]: PEP8

File: Twitter/boxxy.py [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+        # set important vars
+        self.omtrek = int(state["circuitLength"])
+        if self.shortestLapGlobal is None:
+            self.shortestLapGlobal = 5*60  # 5 min
+
+        for teamid in state["teams"]:
+            teamjson = state['teams'][teamid]
+            self.teams[teamid] = Team(teamjson["name"])
+
```
Comment (Silox) [inline Twitter/boxxy.py]: PEP8

File: Twitter/boxxy.py [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+            loc = self.distances.popleft()
+            # create msg
+            andBack = ' (en terug)' if not loc[2] else ''
+            tweet(
+                choice(TOTAL_DISTANCE).format(location=loc[1], andBack=andBack, distance=convert_distance(distance)))
+
+        team.lastLapTimeStamp = lapEndTimeStamp
+
+        self.dump()
+
```
Comment (Silox) [inline Twitter/boxxy.py]: PEP8

File: Twitter/boxxy.py [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
+    def position(self, position):
+        pass
+
```
Comment (Silox) [inline Twitter/boxxy.py]: PEP8

File: Twitter/boxxy.py [hunk 9]
```diff
@@ -0,0 +1,204 @@
...
+    def dump(self):
+        with open(FILENAME, 'w') as f:
+            print(jp.encode(self), end='', file=f)
+
```
Comment (Silox) [inline Twitter/boxxy.py]: PEP8

File: Twitter/boxxy.py [hunk 10]
```diff
@@ -0,0 +1,204 @@
...
+    return str(int(distance/1000)) + "km"
+
+
+def tweet(msg):
+    print("Tweet: " + msg + " (" + str(len(msg)) + ")")
+    try:
+        _ = twitter.statuses.update(status=msg)
+    except:
+        print("Twitter error")
+
```
Comment (Silox) [inline Twitter/boxxy.py]: PEP8

File: Twitter/statejson example [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+        },
+    'stations': {
+        '1': {'name': 'gyrid1', 'id': '1', 'position': 45.0},
+        '2': {'name': 'gyrid2', 'id': '2', 'position': 120.0},
+        '4': {'name': 'gyrid6', 'id': '4', 'position': 280.0},
+        '3': {'name': 'gyrid3', 'id': '3', 'position': 190.0},
+        '5': {'name': 'gyrid5', 'id': '5', 'position': 370.0}
+        },
+    'circuitLength': 380.0, 'startTime': '2015-04-23T23:27:25.683Z'
+}
```
Comment (Silox) [inline Twitter/statejson example]: Newline at end of file

File: Twitter/config.py.example [hunk 12]
```diff
@@ -0,0 +1,4 @@
+CONSUMER_KEY = ''
+CONSUMER_SECRET = ''
+ACCESS_KEY = ''
+ACCESS_SECRET = ''
```
Comment (Silox) [inline Twitter/config.py.example]: Newline at end of file

File: Twitter/boxxy.py [hunk 13]
```diff
@@ -0,0 +1,204 @@
+from socketIO_client import SocketIO, LoggingNamespace
```
Comment (Silox) [inline Twitter/boxxy.py]: Please put these in order:

default libraries
[blank line]
imports from requirements.txt
[blank line]
imports from own code

File: Twitter/boxxy.py [hunk 14]
```diff
@@ -0,0 +1,209 @@
...
+                "{team} {verb} hun persoonlijk record verbeterd: hun laatste rondje duurde slechts {time}. Doe zo voort!"]
+
+GLOBAL_FASTEST_LAP = [
+    "{team} {verb} net een rondje in {time} gelopen. Dat is tot nog toe het snelste rondje van de dag! Proficiat!",
+    "De titel Snelste Rondje Van De Dag gaat momenteel naar {team}. Zij hebben hun laatste rondje in {time} gelopen! Proficiat!",
+    "Een nieuw snelheidsrecord: {team} {verb} hun laatste rondje in {time} gelopen. Dat is tot nu toe het snelste rondje van de dag! Proficiat!"]
+
+TEAM_RUN_ROUNDS = ["{team} {verb} al {laps} laps achter de rug.",
+                   "{team} {verb} al {laps} laps gelopen.",
+                   "Nog eens honderd laps erbij! {team} {verb} al {laps} laps gelopen."]
```
Comment (Iepoev) [inline Twitter/boxxy.py]: s/laps/rondjes/

File: Twitter/boxxy.py [hunk 15]
```diff
@@ -0,0 +1,209 @@
...
+    (2263040, 'Boedapest', False),
+    (2319700, 'Warschau', False),
+    (2342680, 'Rome', False),
+    (2561760, 'Stockholm', False),
+    (2628600, 'Madrid', False),
+    (2745180, 'Belgrade', False),
+    (3395660, 'Sofia', False),
+    (3539380, 'Boekarest', False),
+    (3672400, 'Kiev', False),
+    (3807320, 'Sint', False),
```
Comment (Iepoev) [inline Twitter/boxxy.py]: moet "sint petersburg" zijn

File: Twitter/boxxy.py [hunk 16]
```diff
@@ -0,0 +1,209 @@
...
+    (1372590, 'Belgrade', True),
+    (1393220, 'Milaan', False),
+    (1442160, 'Praag', False),
+    (1533340, 'Kopenhagen', False),
+    (1546400, 'Dublin', False),
+    (1697830, 'Sofia', True),
+    (1769690, 'Boekarest', True),
+    (1829620, 'Wenen', False),
+    (1836200, 'Kiev', True),
+    (1903660, 'Sint', True),
```
Comment (Iepoev) [inline Twitter/boxxy.py]: moet "Sint Petersburg" zijn

**Pull request merged**